prover/lang/kore-lang.md

```diff
@@ -181,13 +181,24 @@ only in this scenario*.
 
   syntax SymbolDeclaration ::= "symbol" Symbol "(" Sorts ")" ":" Sort
   syntax SortDeclaration ::= "sort" Sort
+
+  // defined in `lang/smt-lang.md
+  syntax SMTLIB2Sort 
+  syntax SMTLIB2SimpleSymbol
+
+  syntax HookDeclaration ::= "hook-smt-sort" "(" Sort ","  SMTLIB2Sort ")"
+                           | "hook-smt-symbol" "(" Symbol "," SMTLIB2SimpleSymbol ")"
```

Comment:
Use `syntax Pattern ::= hoot-smt-sort(...)`

so we can do this instead:

```
axiom hook-smt-symbol(store, store)
```

Reply:
I would prefer not to make it a pattern, since it cannot be desugared to normal ML, and we cannot use the hook in other patterns. If you insist on that syntax, I would rather make a new nonterminal `Axiom` that is either `Pattern` or `HookDeclaration`.